HW3_spark_smythe.py

```diff
@@ -1,156 +0,0 @@
-        kwargs = self._input_kwargs
-        self.setParams(**kwargs)
-        return
-
-    @keyword_only
-    def setParams(self, inputCols=None, outputCol=None):
-        kwargs = self._input_kwargs
-        return self._set(**kwargs)
-
-    def _transform(self, dataset):
-        input_cols = self.getInputCols()
-        output_col = self.getOutputCol()
-
-        # Generate a list of columns to append
-        concat_list = []
-        for count, column in enumerate(input_cols):
-            column_func = when(col(column).isNull(), lit("")).otherwise(col(column))
-            concat_list.append(column_func)
-            if count < len(input_cols) - 1:
-                concat_list.append(lit(" "))
-        dataset = dataset.withColumn(
-            output_col, split(concat(*concat_list), pattern=" ")
-        )
-        return dataset
-
-
-def main():
-    # Setup Spark
-    spark = SparkSession.builder.master("local[*]").getOrCreate()
-
-    # set up sql defaults
-    database = "baseball"
-    port = "3306"
-    user = "root"
-    password = "x11desktop"  # pragma: allowlist secret
-
-    # get table and make batter df
-    bb_df = (
-        spark.read.format("jdbc")
-        .options(
-            url=f"jdbc:mysql://localhost:{port}/{database}",
-            driver="com.mysql.cj.jdbc.Driver",
-            dbtable="batter_counts",
-            user=user,
-            password=password,
-            inferSchema="true",
-            header="true",
-        )
-        .load()
-    )
-
-    bb_df.select("batter", "game_id", "atBat", "Hit").show()
-    bb_df.createOrReplaceTempView("batter")
-    bb_df.persist(StorageLevel.DISK_ONLY)
-
-    # get table and make game df
-    gd_df = (
-        spark.read.format("jdbc")
-        .options(
-            url=f"jdbc:mysql://localhost:{port}/{database}",
-            driver="com.mysql.cj.jdbc.Driver",
-            dbtable="game",
-            user=user,
-            password=password,
-            inferSchema="true",
-            header="true",
-        )
-        .load()
-    )
-
-    gd_df.select("game_id", "local_date").show()
-    gd_df.createOrReplaceTempView("game")
-    gd_df.persist(StorageLevel.DISK_ONLY)
-
-    bat_avg = spark.sql(
-        """
-        select
-            batter
-            , atBat
-            , Hit 
-            , (Hit/nullif(atBat,0)) as game_bat_avg
-        from batter
-        """
-    )
-    bat_avg.show()
-    # make pipeline
-
-    split_column_transform = SplitColumnTransform(
-        inputCols=["batter", "game_id", "atBat", "Hit"], outputCol="categorical"
-    )
-    count_vectorizer = CountVectorizer(
-        inputCol="categorical", outputCol="categorical_vector"
-    )
-    random_forest = RandomForestClassifier(
-        labelCol="batter",
-        featuresCol="categorical_vector",
-        numTrees=100,
-        predictionCol="batting_avg",
-        probabilityCol="prob_of_hit",
-        rawPredictionCol="raw_pred_hit",
-    )

```

Comment:
RandomForestClassifer isn't needed in this hw assignment as far as I know. In fact I don't think it will work in this case either. RandomForestClassifer is used when you want to predict 0 or 1 as the target but you put the target or label as "batter". The "batter" isn't a binary output.

Reply:
Ah i missed this on this version. pre revert i had it for hit. to predict if there would be a hit. I will have to track down where i missed.